rust/kernel/src/miscdev.rs

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use crate::error::{Error, KernelResult};
+use crate::file_operations::{FileOperations, FileOperationsVtable};
+use crate::{bindings, c_types, CStr};
+use core::marker::PhantomPinned;
+use core::pin::Pin;
+
+/// A registration of a misc device.
+pub struct Registration {
+    mdev: bindings::miscdevice,
+    registered: bool,
```

Comment:
Can this just be an `Option<miscdevice>` instead of a value + bool?

Reply:
Agreed -- perhaps Wedson expects more fields in `Registration`?

Reply:
Yes, it can. Changing it.